lib/matplotlib/image.py

```diff
@@ -414,12 +425,37 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 # Convert back to a masked greyscale array so
                 # colormapping works correctly
                 hid_output = output
+                # any pixel where the a masked pixel is included
+                # in the kernel (pulling this down from 1) needs to
+                # be masked in the output
+                if len(mask.shape) == 2:
+                    out_mask = np.empty((out_height, out_width),
+                                        dtype=mask.dtype)
+                    _image.resample(mask, out_mask, t,
+                                    _interpd_[self.get_interpolation()],
+                                    True, 1,
+                                    self.get_filternorm() or 0.0,
+                                    self.get_filterrad() or 0.0)
+                    out_mask = np.isnan(out_mask)
+                else:
+                    out_mask = mask
+                # we need to mask both pixels which came in as masked
+                # and the pixels that Agg is telling us to ignore (relavent
+                # to non-affine transforms)
+                # Use half alpha as the threshold for pixels to mask.
+                out_mask = out_mask | (hid_output[..., 3] < .5)
                 output = np.ma.masked_array(
-                    hid_output[..., 0], hid_output[..., 3] < 0.5)
-                # relabel under data
-                output[hid_output[..., 1] > .5] = -1
+                    hid_output[..., 0],
+                    out_mask)
+                # 'unshare' the mask array to
+                # needed to suppress numpy warning
+                del out_mask
+                invalid_mask = ~output.mask * ~np.isnan(output.data)
```

Comment:
Are there performance rather than semantic advantages?  I think `~(a & b)` would also work and have one less temprorary.